### PR TITLE
Allow Markdown and description placeholder usage in data field descriptions

### PR DESCRIPTION
**Pull request by frenck**
## Breaking change

Before:

image

After:

CleanShot 2022-08-13 at 14 52 06

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/dialogs/config-flow/show-dialog-config-flow.ts [hunk 1]
```diff
@@ -89,9 +89,13 @@ export const showConfigFlowDialog = (
...
 
     renderShowFormStepFieldHelper(hass, step, field) {
-      return hass.localize(
-        `component.${step.handler}.config.step.${step.step_id}.data_description.${field.name}`
+      const description = hass.localize(
+        `component.${step.handler}.config.step.${step.step_id}.data_description.${field.name}`,
+        step.description_placeholders
       );
+      return description
+        ? html` <ha-markdown breaks .content=${description}></ha-markdown> `
```
Comment (zsarnett) [inline src/dialogs/config-flow/show-dialog-config-flow.ts:97]:
```suggestion
        ? html`<ha-markdown breaks .content=${description}></ha-markdown>`
```

File: src/dialogs/config-flow/show-dialog-data-entry-flow.ts [hunk 2]
```diff
@@ -61,7 +61,7 @@ export interface FlowConfig {
     hass: HomeAssistant,
     step: DataEntryFlowStepForm,
     field: HaFormSchema
-  ): string;
+  ): TemplateResult | "";
```
Comment (zsarnett) [inline src/dialogs/config-flow/show-dialog-data-entry-flow.ts:64]:
```suggestion
  ): TemplateResult | string;
```

File: src/dialogs/config-flow/show-dialog-options-flow.ts [hunk 3]
```diff
@@ -93,9 +93,13 @@ export const showOptionsFlowDialog = (
...
 
       renderShowFormStepFieldHelper(hass, step, field) {
-        return hass.localize(
-          `component.${configEntry.domain}.options.step.${step.step_id}.data_description.${field.name}`
+        const description = hass.localize(
+          `component.${configEntry.domain}.options.step.${step.step_id}.data_description.${field.name}`,
+          step.description_placeholders
         );
+        return description
+          ? html` <ha-markdown breaks .content=${description}></ha-markdown> `
```
Comment (zsarnett) [inline src/dialogs/config-flow/show-dialog-options-flow.ts:101]:
```suggestion
          ? html`<ha-markdown breaks .content=${description}></ha-markdown>`
```

File: src/panels/config/repairs/show-dialog-repair-flow.ts [hunk 4]
```diff
@@ -100,11 +100,15 @@ export const showRepairsFlowDialog = (
...
-        return hass.localize(
+        const description = hass.localize(
           `component.${issue.domain}.issues.${
             issue.translation_key || issue.issue_id
-          }.fix_flow.step.${step.step_id}.data_description.${field.name}`
+          }.fix_flow.step.${step.step_id}.data_description.${field.name}`,
+          step.description_placeholders
         );
+        return description
+          ? html` <ha-markdown breaks .content=${description}></ha-markdown> `
```
Comment (zsarnett) [inline src/panels/config/repairs/show-dialog-repair-flow.ts:110]:
```suggestion
          ? html`<ha-markdown breaks .content=${description}></ha-markdown>`
```

**Review by zsarnett — DISMISSED** [review]

**Review by zsarnett — DISMISSED** [review]

**Review by zsarnett — APPROVED** [review]

**Comment by frenck** [comment]
Thanks for finishing @zsarnett ❤️

**Pull request merged**